modules/jdbc/src/main/java/org/testcontainers/containers/JdbcDatabaseContainer.java

```diff
@@ -60,6 +60,10 @@ public JdbcDatabaseContainer(String dockerImageName) {
      */
     protected abstract String getTestQueryString();
 
+    public abstract SELF withUsername(String username);
```

Comment:
Since we are missing a default implementation here, this will break API compatibility with other JDBC modules, like https://github.com/testcontainers/testcontainers-java-module-mssqlserver. What do you think about implementing a default method throwing `UnsupportedOperationException` instead of abstrac method?

I see you already did this for Virtuso, just wondering how we can easily align the external modules.